js/hview.js

```diff
@@ -667,150 +669,155 @@ $(function () {
+            w.terminate()
+            window.removeEventListener('popstate', clearState)
+        }
+
+        // no memory leaks if the time lapse data view is dismissed.
+        window.addEventListener('popstate', clearState)
+
+        function removeChildBoxes(oldNode) {
+            oldNode.children.forEach(it => {
+                removeChildBoxes(it)
+                document.getElementById(it.treeDisplayName).box.remove()
+            })
+        }
+
+        function onSwitch(oldIndex, newIndex) {
+            const treeDifferences = compareNodes(rootNodes[newIndex], rootNodes[oldIndex])
+
+            // delete all children, then add them again properly
+            treeDifferences.childUpdates.forEach(it => {
+                const node = viewNode_deepCloneOf(it)
+                const el = document.getElementById(node.treeDisplayName)
+                removeChildBoxes(el.node)
+
+                node.el = el
+                node.box = el.box
+                el.node = node
+
+                const parentEl = document.getElementById(node.parent.treeDisplayName)
+                parentEl.node.children[indexOfChild(parentEl.node, node)] = node
+                node.parent = parentEl.node
+
+                if (node.children.length > 0) {
+                    if (!el.childContainer) {
+                        el.classList.add(CLS_EXPANDABLE)
+                        el.ondblclick = treeToggle
+                        el.childContainer = newContainerProtoType.cloneNode()
+                        // insertBefore adds to the end if el.nextSibling is null
+                        el.parentNode.insertBefore(el.childContainer, el.nextSibling)
+                    } else {
+                        el.childContainer.replaceChildren()
+                    }
+                    const boxContainer = document.getElementById("border-box")
+                    node.children.forEach(child => {
+                        renderNode(child, el.childContainer, boxContainer)
+                    })
+                } else {
+                    if (el.childContainer) {
+                        el.classList.remove(CLS_EXPANDABLE)
+                        el.ondblclick = null
+                        el.childContainer.remove()
+                        el.childContainer = null
+                    }
+                }
+            })
+            if (treeDifferences.childUpdates.length > 0) showHiddenNodeOptionChanged()
+
+            // move the box position in the image preview
+            treeDifferences.boxPosUpdates.forEach(node => {
+                const el = document.getElementById(node.treeDisplayName)
+                el.box.style.left = node.boxStylePos.left
+                el.box.style.top = node.boxStylePos.top
+                el.box.style.width = node.boxStylePos.width
+                el.box.style.height = node.boxStylePos.height
+            })
+
+            // if selected node's properties changed, update the property window
```

Comment:
This section on changing properties could be simplified/have the optimization removed to just "re-render the properties screen" if you think the following code is too complex. I opted to leave it here since it is faster, but wanted to make note of this since we unoptimized / simplifying the adding/removing node children.